### PR TITLE
Migrate SamplingDeviations to DX

### PR DESCRIPTION
**Pull request by Bugerman58**
## Description of the issue/feature this PR addresses

This PR migrates the AT SamplingDeviation and SamplingDeviations to Dexterity types for Python 3 compatibility.

## Current behavior before PR

SamplingDeviation and SamplingDeviations are content types based on the Archetypes framework

## Desired behavior after PR is merged

SamplingDeviation and SamplingDeviations are content types based on the Dexterity framework

--
I confirm I have tested this PR thoroughly and coded it according to [PEP8][1]
and [Plone's Python styleguide][2] standards.

[1]: https://www.python.org/dev/peps/pep-0008
[2]: https://docs.plone.org/develop/styleguide/python.html

### CODE REVIEW

**Comment by Bugerman58** [comment]
Hi @xispa this PR ready for review.

File: src/senaite/core/browser/controlpanel/samplingdeviations/view.py [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+                "transitions": [{"id": "empty"}, ],
+                "columns": self.columns.keys(),
+            }, {
+                "id": "active",
+                "title": _(
+                    "listing_samplingdeviations_state_active",
+                    default="Active",
+                ),
+                "contentFilter": {"is_active": True},
+                "transitions": [{"id": "deactivate"}, ],
```
Comment (xispa) [inline src/senaite/core/browser/controlpanel/samplingdeviations/view.py]: Remove this `transitions` attribute. Listing displays the available transitions for the selected object automatically based on the workflow(s) the object is bound to.

File: src/senaite/core/browser/controlpanel/samplingdeviations/view.py [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+                "columns": self.columns.keys(),
+            }, {
+                "id": "inactive",
+                "title": _(
+                    "listing_samplingdeviations_state_inactive",
+                    default="Inactive",
+                ),
+                "contentFilter": {'is_active': False},
+                "transitions": [{"id": "activate"}, ],
```
Comment (xispa) [inline src/senaite/core/browser/controlpanel/samplingdeviations/view.py]: Remove `transitions` here too

File: src/senaite/core/browser/controlpanel/samplingdeviations/view.py [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+
+        self.review_states = [
+            {
+                "id": "default",
+                "title": _(
+                    "listing_samplingdeviations_state_all",
+                    default="All",
+                ),
+                "contentFilter": {},
+                "transitions": [{"id": "empty"}, ],
```
Comment (xispa) [inline src/senaite/core/browser/controlpanel/samplingdeviations/view.py]: Remove this `transitions` attribute cause default transitions "Activate" (if deactivated) or "Deactivate" (if active) are not displayed when "All" filter is selected

**Review by xispa — CHANGES_REQUESTED** [review]
Excellent!, just three small things left!

**Review by xispa — APPROVED** [review]

**Pull request merged**